common/Resources/ti.internal/extensions/process.js

```diff
@@ -46,14 +46,87 @@ function standardizeArch(original) {
 	}
 }
 
+const listeners = {};
 const process = {
 	arch: standardizeArch(Ti.Platform.architecture),
+	noDeprecation: false,
+	throwDeprecation: false,
+	traceDeprecation: false,
+	pid: 0,
 	cwd: function () {
 		return __dirname;
 	},
 	// FIXME: Should we try and adopt 'windowsphone'/'windowsstore' to 'win32'?
 	// FIXME: Should we try and adopt 'ipad'/'iphone' to 'darwin'? or 'ios'?
-	platform: Ti.Platform.osname
+	platform: Ti.Platform.osname,
+	on: function (eventName, callback) {
+		const eventListeners = listeners[eventName] || [];
+		eventListeners.push(callback);
+		listeners[eventName] = eventListeners;
+		return this;
+	},
+	// TODO: Add #once which is like #on, but should get wrapped to remove itself before getting fired
+	emit: function (eventName, ...args) {
+		const eventListeners = listeners[eventName] || [];
+		for (const listener of eventListeners) {
+			listener.call(this, ...args);
+		}
+		return eventListeners.length !== 0;
+	},
+	eventNames: () => Object.getOwnPropertyNames(listeners),
+	emitWarning: function (warning, options, code, ctor) {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-default;
  File: common/Resources/ti.internal/extensions/process.js;
  Line: 77;
-->

- :warning: common/Resources/ti.internal/extensions/process.js line 77 – 'ctor' is defined but never used. (no-unused-vars)